### DISCUSSION TITLE
General Approach for Sketching Implicit Functions (e.g., Circles)

### DISCUSSION THREAD

**Discussion by michelmar** [stevenpetryk/mafs#122] [Ideas]
I'm looking for a good approach to sketch implicit functions, such as (x-x0)^2 + (y-y0)^2 = R^2, which represents a circle. I am aware that Mafs has a Circle element, but I am interested in a more general solution that can handle a variety of implicit equations.

To Reproduce:

Currently, I am unable to plot implicit functions using Mafs, other than specific cases like circles using the Circle element, or Elipsis.

Expected behavior:

I would like to have a method or functionality to plot implicit functions in a more general manner, not limited to circles or ellipsis. This would enable users to visualize different types of equations, enhancing the capabilities of the library.

Any guidance or recommendations for implementing such a feature would be greatly appreciated!

**Comment by stevenpetryk** [comment]
Just following up on this, a little late! I also converted it to a discussion. 

We've had some discussion [in Discord](https://discord.gg/mafs) around implicit function plotting. The general outcome is that, well, it's tricky from a performance standpoint in JS-world. However, @shovon has written some interesting code in [shovon/implicit-quad-tree](https://github.com/shovon/implicit-quad-tree).

I think it would be relatively simple to convert this to Mafs-compatible code, since the way Mafs handles plotting right now is just to render an SVG `path` element.
